sympy/core/tests/test_expr.py

```diff
@@ -215,13 +215,14 @@ def test_relational_assumptions():
     assert (m4 >= 0) is S.false
 
 
-def test_relational_noncommutative():
-    from sympy import Lt, Gt, Le, Ge
-    A, B = symbols('A,B', commutative=False)
-    assert (A < B) == Lt(A, B)
-    assert (A <= B) == Le(A, B)
-    assert (A > B) == Gt(A, B)
-    assert (A >= B) == Ge(A, B)
+# See https://github.com/sympy/sympy/issues/17708
+#def test_relational_noncommutative():
+#    from sympy import Lt, Gt, Le, Ge
+#    A, B = symbols('A,B', commutative=False)
+#    assert (A < B) == Lt(A, B)
+#    assert (A <= B) == Le(A, B)
+#    assert (A > B) == Gt(A, B)
+#    assert (A >= B) == Ge(A, B)
```

Comment:
There are still some things to address for this, like a misleading error message
```python3
>>> from sympy import *
>>> A, B = symbols('A,B', commutative=False)
>>> A < B
```
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-ec72fe6fe21f> in <module>
      2 
      3 A, B = symbols('A,B', commutative=False)
----> 4 A < B

~\Documents\Github\sympy\sympy\core\expr.py in __lt__(self, other)
    408         for me in (self, other):
    409             if me.is_extended_real is False:
--> 410                 raise TypeError("Invalid comparison of complex %s" % me)
    411             if me is S.NaN:
    412                 raise TypeError("Invalid NaN comparison")

TypeError: Invalid comparison of complex A
```